src/App.ts

```diff
@@ -886,7 +887,8 @@ export default class App {
                 context,
                 client,
                 logger: this.logger,
-              }),
+                // `next` is already set in the outer processMiddleware
```

Comment:
The code here is a bit tricky. You can check the internals of `processMiddleware`. The outer `processMiddleware` call quietly sets `next` and middleware execution relies on it.

Reply:
Are the comments on 884 and 871 technically misleading then, if `next` is actually set?

Reply:
@srajiang Indeed, `last()` and L871 may be confusing. In my understanding, the last element of an array of middleware technically can have `next()` but we don't do anything with it. As you saw, inside the `processMiddleware` method, we call `last()` instead of next middleware with recursive `invokeMiddleware` call. 

We may want to update the L871 comment when merging this PR.

Reply:
Updated the comment